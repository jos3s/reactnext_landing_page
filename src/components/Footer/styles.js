import styled from 'styled-components';

export const Container = styled.footer`
  text-align: center;
  a {
    color: inherit;
    text-decoration: none;
    font-size: ${({ theme }) => theme.font.sizes.small};
  }
`;
