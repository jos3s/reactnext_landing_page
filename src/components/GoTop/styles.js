import styled from 'styled-components';

export const Container = styled.a`
  position: fixed;
  background: ${({ theme }) => theme.colors.primaryColor};
  color: ${({ theme }) => theme.colors.white};
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
`;
