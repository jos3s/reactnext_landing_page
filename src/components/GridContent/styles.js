import styled from 'styled-components';

export const Container = styled.div`
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
`;

export const Html = styled.div`
  margin: ${({ theme }) => theme.spacings.large} 0;
`;
