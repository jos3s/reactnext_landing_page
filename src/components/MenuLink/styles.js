import styled from 'styled-components';

export const Container = styled.a`
  color: ${({ theme }) => theme.colors.primaryColor};
  font-size: ${({ theme }) => theme.font.sizes.small};
  padding: ${({ theme }) => theme.spacings.small};
  text-decoration: none;
  display: block;
  position: relative;

  &::after {
    content: '';
    position: absolute;
    bottom: 0.76rem;
    width: 0;
    left: 50%;
    height: 0.2rem;
    background: ${({ theme }) => theme.colors.secondaryColor};
    transition: all 300ms ease-in-out;
  }

  &:hover::after {
    width: 50%;
    left: 25%;
  }
`;
