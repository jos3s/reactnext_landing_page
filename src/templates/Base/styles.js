import styled from 'styled-components';

export const Container = styled.div`
  padding-top: 5.4rem;

  @media ${({ theme }) => theme.media.lteMedium} {
    padding-top: 0;
  }
`;
