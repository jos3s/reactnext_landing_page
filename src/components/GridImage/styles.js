import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../Heading/styles';

export const Container = styled.div`
  ${TextComponent} {
    margin-bottom: ${({ theme }) => theme.spacings.xhuge};
  }
`;

export const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: ${({ theme }) => theme.spacings.large};
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms easy-in-out;

  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
