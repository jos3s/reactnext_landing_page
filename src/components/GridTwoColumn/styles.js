import styled from 'styled-components';

import { Title } from './../Heading/styles';

export const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr 2fr;
  align-items: center;
  gap: ${({ theme }) => theme.spacings.large};

  @media ${({ theme }) => theme.media.lteMedium} {
    grid-template-columns: 1fr;
    text-align: center;
  }

  ${Title} {
    margin-bottom: ${({ theme }) => theme.spacings.xlarge};
  }
`;

export const TextContainer = styled.div`
  @media ${({ theme }) => theme.media.lteMedium} {
    margin-bottom: ${({ theme }) => theme.spacings.large};
  }
`;

export const ImageContainer = styled.div``;

export const Image = styled.img`
  width: 100%;
`;
