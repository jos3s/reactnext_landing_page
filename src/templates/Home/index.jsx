import { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';

import mockBase from '../Base/mock';
import { mapData } from '../../api/mapData';

import { Base } from '../Base';
import { PageNotFound } from '../PageNotFound';
import { Loading } from '../Loading';
import { GridTwoColumn } from '../../components/GridTwoColumn';
import { GridContent } from '../../components/GridContent';
import { GridText } from '../../components/GridText';
import { GridImage } from '../../components/GridImage';

function Home() {
  const [data, setData] = useState([]);

  const location = useLocation();

  useEffect(() => {
    const path = location.pathname.replace(/[^a-z0-9-_]/gi, '');
    const slug = path ? path : 'landing-page';

    const load = async () => {
      try {
        const data = await fetch('http://localhost:1337/pages/?slug=' + slug);
        const json = await data.json();
        const pagesData = mapData(json);
        setData(pagesData[0]);
      } catch (e) {
        console.error(e);
        setData(undefined);
      }
    };
    load();
  }, [location]);

  if (data === undefined) return <PageNotFound />;

  if (data && !data.slug) return <Loading />;

  const { sections, menu, footerHtml, slug } = data;
  const { links, text, link, srcImg } = menu;

  return (
    <Base
      links={links}
      footerHtml={footerHtml}
      logoData={{ text, link, srcImg }}
    >
      {sections.map((section, idx) => {
        const { component } = section;
        const key = `${slug}-${idx}`;
        if (component === 'section.section-two-columns')
          return <GridTwoColumn {...section} key={key} />;
        if (component === 'section.section-content')
          return <GridContent {...section} key={key} />;
        if (component === 'section.section-grid-text')
          return <GridText {...section} key={key} />;
        if (component === 'section.section-grid-image')
          return <GridImage {...section} key={key} />;
      })}
    </Base>
  );
}

export default Home;
