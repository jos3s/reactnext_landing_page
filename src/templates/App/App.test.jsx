import { screen } from '@testing-library/react';
import Home from '.';
import { renderTheme } from '../../styles/renderTheme';

test('renders learn react link', () => {
  renderTheme(<Home />);
  const headingContainer = screen.getByRole('heading', { name: 'Hello World' }).parentElement;
  expect(headingContainer).toHaveStyle({
    background: 'tomato',
  });
  expect(headingContainer).toMatchSnapshot();
  expect(headingContainer).toHaveStyleRule('background', 'tomato');
});
