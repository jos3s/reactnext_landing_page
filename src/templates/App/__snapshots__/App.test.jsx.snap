// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
.c0 {
  background: tomato;
  color: white;
}

<div
  class="c0"
  color="white"
>
  <h1>
    Hello World
  </h1>
</div>
`;
